### PR TITLE
feat(button): change type to kind

### PR DESCRIPTION
**Pull request by kahwee**
* refactor `$type` to `$kind`
* add compact `$size`
* more examples

I didn't get $kind and $size to work as expected. It seems that `styled` didn't persist the props from overridden BaseButton. Will appreciate if you can take a look.

### CODE REVIEW

File: src/components/button/button.js [hunk 1]
```diff
@@ -14,14 +14,21 @@ export default function Button({
   startEnhancer,
   endEnhancer,
   overrides = {},
+  $size,
+  $kind,
```
Comment (schnerd) [inline src/components/button/button.js]: We only use the $ prefix on styled components. For the main exports you should just use normal props `size` and `kind`. When they are passed to "BaseButton", "ButtonLabel", etc, that's when we prefix with a dollar.

File: src/components/button/styled-components.js [hunk 2]
```diff
@@ -1,30 +1,38 @@
...
+  ...$theme.typography[$size === BUTTON_SIZE.compact ? 'font200' : 'font300'],
   alignItems: 'center',
   justifyContent: 'space-between',
   border: 'none',
-  borderRadius: $type === BUTTON_TYPE.round ? '50%' : $theme.borders.radius200,
+  borderRadius: $kind === BUTTON_KIND.round ? '50%' : $theme.borders.radius200,
   outline: 'none',
   WebkitAppearance: 'none',
+  transitionDuration: $theme.animation.timing100,
+  transitionTimingFunction: $theme.animation.easeOutCurve,
```
Comment (schnerd) [inline src/components/button/styled-components.js]: Should we be explicit about which properties are being transitioned? (background color, etc).

Comment (kahwee) [reply src/components/button/styled-components.js]: Good point, I think it should only be the background color. I thought the immediate change of background color looks too harsh.

File: src/components/button/button.js (hunk 1 above)
Comment (kahwee) [reply src/components/button/button.js]: Gotcha, updating to address this.

**Comment by abmai** [comment]
Looks good! I think once this is merged you can just run with whatever's on `anh.button` and work on it. It's missing flow types (cuz I was speeding), and proper component overrides usage iirc.

Thanks, @kahwee!!

**Pull request merged**